src/dmd/parse.d

```diff
@@ -5401,7 +5386,7 @@ class Parser(AST) : Lexer
      * If `isStatic` is true, `isDecl` can be true to indicate that a
      * `static foreach` declaration should be parsed.
      */
-    private ParseForeachRet!(isStatic, isDecl) parseForeach(bool isStatic, bool isDecl)(Loc loc, ParseForeachArgs!(isStatic, isDecl) args)
+    private ParseForeachRet!(isStatic, isDecl) parseForeach(bool isStatic, bool isDecl)(Loc loc, AST.Dsymbol* pLastDecl)
```

Comment:
Please update the DDoc to specify what `pLastDecl` is and that if `isDecl` is `false`, then `null` should be passed to `pLastDecl`.

Reply:
This is covered in the followup to this PR https://github.com/dlang/dmd/pull/12922